proposals/custom-element-default-focus.md

```diff
@@ -36,13 +37,24 @@ class MyButton extends HTMLElement {
 
 ## Proposal
 
-Add a new property `matchFocusBehaviorOf` to `ElementInternals`, which accepts a string containing a tagname and optionally an attribute with a value (so it will accept values like `div`, `a[href]`, `input[type='checkbox']` , but not `.class`, `*`, etc.). The custom element will then be treated as if it has that tagname and attribute in terms of focus.
+Add `focusBehavior` property to `ElementInternals`, which accepts one of three string values; `"none"`, `"focusable"`, and `"infer"`.
+`"none"` indicates the element is not focusable, and `"focusable"` indicates the element's focus behaivor is same as `<div tabindex=non-negative-value>`. `"infer"` indicates the element is focusable, however a user-agent decides click behaivor and TAB behavior with hints from the element.
```

Comment:
So we are excluding programmatically focusable from the list of options?

Reply:
I thought programatically-focusable was always true for `"focusable"` and `"infer"`.  The last sentence should mention programatically-focusable if such a built-in element exists.

Reply:
Apologies, I should have said programmatically-focusable *only*.

Reply:
> programmatically-focusable only.

Ah, I understand your concern.  [The last list in your comment](https://github.com/w3c/webcomponents/issues/762#issuecomment-542509589) includes "programatically focusable".

I dropped it because I was not sure if there are use-cases of the explicit programatically-focusable-only, and I thought following an existing built-in element with programatically-focusable-only by `"infer"` would be enough.

Anyway, I'd like to update [the issue thread](https://github.com/w3c/webcomponents/issues/762) with this information.